packages/design-toolkit/src/components/flashcard/index.tsx

```diff
@@ -0,0 +1,200 @@
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/**
+ * Example usage.
+ *
+ * <Flashcard>
+ *  <FlashcardHero>
+ *    <FlashcardIdentifier>
+ *      <FlashcardHeader>
+ *        {header}
+ *      </FlashcardHeader>
+ *      <FlashcardSubheader>
+ *        {subHeader}
+ *      </FlashcardSubheader>
+ *    </FlashcardIdentifier>
+ *  </FlashcardHero>
+ *  <FlashcardSecondary>
+ *    <FlashcardSecondaryData>
+ *      {secondaryData}
+ *    </FlashcardSecondaryData>
+ *    <FlashcardDetailsContainer details={details} />
+ *  </FlashcardSecondary>
+ * </Flashcard>
+ */
+'use client';
+import 'client-only';
+import { createContext, useContext } from 'react';
+import { Skeleton } from '../skeleton';
+import { FlashcardStyles } from './styles';
+import type {
+  FlashcardComponentProps,
+  FlashcardDetailContainerProps,
+  FlashcardProps,
+} from './types';
+
+const {
+  container,
+  hero,
+  identifier,
+  header,
+  subHeader,
+  secondaryContainer,
+  secondaryData,
+  detailsContainer,
+  detailsItem,
+  detailsLabel,
+  detailsValue,
+  skeleton,
+} = FlashcardStyles();
+
+export const FlashcardContext = createContext<FlashcardProps>({
+  isLoading: false,
+});
+
+export function Flashcard(props: FlashcardProps) {
+  const { isLoading, children, className, ...rest } = props;
+
+  return (
+    <FlashcardContext.Provider value={{ isLoading }}>
+      <div {...rest} className={container({ className })}>
+        {children}
+      </div>
+    </FlashcardContext.Provider>
+  );
+}
+Flashcard.displayName = 'Flashcard';
+
+export function FlashcardHero(props: FlashcardComponentProps) {
+  const { children, className, ...rest } = props;
+  const { isLoading } = useContext(FlashcardContext);
+
+  if (isLoading) {
+    return (
+      <div {...rest} className={hero({ className })}>
+        <Skeleton className={skeleton()} />
+        <Skeleton className={skeleton({ className: 'max-w-1/2' })} />
+      </div>
+    );
+  }
+
+  return (
+    <div {...rest} className={hero()}>
+      {children}
+    </div>
+  );
+}
+FlashcardHero.displayName = 'FlashcardHero';
+
+export function FlashcardIdentifier(props: FlashcardComponentProps) {
+  const { children, className, ...rest } = props;
+
+  return (
+    <div {...rest} className={identifier({ className })}>
+      {children}
+    </div>
+  );
+}
+FlashcardIdentifier.displayName = 'FlashcardIdentifier';
+
+export function FlashcardHeader(props: FlashcardComponentProps) {
```

Comment:
You can simplify this component to just be `export const FlashcardHeader = Heading`, then the subheader can be `export const FlashcardSubheader = Text`, and then implement the context provider in the hero to pass the classes down. This helps to eliminate wrapper components that do almost nothing